openshift/machine-config-operator
# OCPBUGS-8403 Deleting SSH keys / password hashes should not degrade MachineConfigPool / node

**- What I did**
Removed conditional that triggers degradation if a core user is not supplied. Added conditional that verifies user data before ssh update.

**- How to verify it**
1. create a cluster without uploading an ssh key
2. oc apply the machine config with an ssh key 
3. ensure that the nodes and mcp update properly without degradation
4. oc mc delete the matching config with the ssh key
5. ensure again that the nodes and mcp update properly without degradation

Do the above again but with a password hash set. ensure that the password is cleared from /etc/shadow


**- Description for the changelog**
